### ISSUE TITLE
Running `npx vitepress` on clean installation results in "failed to start server. error: SyntaxError: Unexpected end of JSON input"

### ISSUE THREAD

**Issue by revosw** [vuejs/vitepress#1112]
### Describe the bug

When any `vitepress` command is run on a clean installation of vitepress+vue using npm, it results in the error below
```
vitepress v1.0.0-alpha.4
failed to start server. error:
 SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at hasWorkspacePackageJSON (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:60067:26)
    at searchForWorkspaceRoot (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:60095:9)
    at searchForWorkspaceRoot (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:60101:12)
    at initTSConfck (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:19923:27)
    at Object.configResolved (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:19813:19)
    at C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:61632:139
    at Array.map (<anonymous>)
    at resolveConfig (C:\dev\js\project\node_modules\vite\dist\node\chunks\dep-c9998dc6.js:61632:47)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### Reproduction

Create a new folder called `project` anywhere on the machine, and navigate inside the folder

`npm init -y`

Create a `docs` folder with a file `index.md` inside it. Write `# hello` inside the file and save.

The file structure now looks like this
```
project/
├─ docs/
│  ├─ index.md
├─ package.json
```

Run `npx vitepress build` in the terminal

### Expected behavior

The `npx vitepress` set of commands run successfully

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i5-12600
    Memory: 49.05 GB / 63.76 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.12.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (103.0.1264.77)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    vitepress: ^1.0.0-alpha.4 => 1.0.0-alpha.4
```

### Additional context

Contents of package.json
```json
{
  "name": "project",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "vitepress": "^1.0.0-alpha.4",
    "vue": "^3.2.37"
  }
}
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Have you added `vitepress` and `vue` to your dependencies?

**Comment by revosw** [comment]
@brc-dd Yes, I have added both vitepress and vue to devDependencies, ref. Additional context section in my bug report

Interestingly, this error does not happen when I do the same on my laptop, so there is something fundamentally wrong on my desktop (the machine I reported the bug on). I reinstalled Nodejs on the desktop, but I still get the same error. If I find exactly why I get the error, I'll update the post and close the issue
